### PR TITLE
[1317] Add edit provider name feature and summary list for providers

### PR DESCRIPTION
**Pull request by JonF1**
### Context
https://trello.com/c/cwKSUfTf/1317-s-add-edit-provider-name-feature-to-providers-pages

### Changes proposed in this pull request
* Adds summary list component with change link.  Added name, code and last updated but can swap if other fields wanted. Link takes the user to form where they can change name. Cancel option to return if needed. Last updated date uses gov_uk format.
* Uniqueness and length validations. I set 100 chars limit as no provider name in db is more than 80 chars 
* Feature test
### Guidance to review
Go to `/support/providers` and select a provider
You may have to comment out `before_action :authenticate` in application controller

Screenshot 2021-04-13 at 15 47 18
Screenshot 2021-04-08 at 17 08 04

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased master
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

**Comment by JonF1** [comment]
[[S] Add "Edit provider name" feature to providers' pages](https://trello.com/c/cwKSUfTf/1317-s-add-edit-provider-name-feature-to-providers-pages)

File: app/webpacker/styles/application.scss [hunk 1]
```diff
@@ -1,3 +1,12 @@
...
 @import "~@ministryofjustice/frontend/moj/all";
+
+.app-provider-name {
+   display: inline;  
+   .app-provider-edit-link {
+    @include govuk-font($size: 19);
+     display: inline;
+     padding: govuk-spacing(2)
+   }
+}
```
Comment (darokel) [inline app/webpacker/styles/application.scss:12]: line endings 🙈

**Review by darokel — COMMENTED** [review]
Looking good. I wonder if we should just follow this pattern to allow names to be changed:

Screenshot 2021-04-09 at 16 01 43

We should probably start populating this summary view in the details tab and show their name there with the change link to the right? This would match our pattern in register.

**Comment by JonF1** [comment]
> We should probably start populating this summary view in the details tab and show their name there with the change link to the right? This would match our pattern in register.

Ok I'll add blocked to the ticket

File: app/webpacker/styles/application.scss (hunk 1 above)
Comment (JonF1) [reply app/webpacker/styles/application.scss]: i fixed by removing all the css 😆

File: app/views/support/providers/edit.html.erb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+      
+  <%= f.govuk_error_summary %>
+
+  <%= f.govuk_text_field :provider_name, label: { text: "New Provider name", size: "s" }, width: 20 %>
+  <%= f.govuk_submit %>
+  <% end %>
+
+  <%= govuk_link_to(t("cancel"), support_provider_path) %>
+  </div>
+</div>
```
Comment (darokel) [inline app/views/support/providers/edit.html.erb:14]: line ending

File: app/views/support/providers/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    key: 'Code',
+    value: @provider.provider_code
+  )
+
+  component.slot(
+    :row,
+    key: 'Last updated',
+    value: @last_updated
+  )
+end %>
```
Comment (darokel) [inline app/views/support/providers/show.html.erb:20]: line ending

File: app/views/support/providers/show.html.erb [hunk 4]
```diff
@@ -0,0 +1,20 @@
+<%= render GovukComponent::SummaryList.new do |component|
+  component.slot(
+    :row,
+    key: 'Name',
```
Comment (darokel) [inline app/views/support/providers/show.html.erb:4]: could we swap for double quotes

File: app/webpacker/styles/application.scss [hunk 5]
```diff
@@ -1,3 +1,3 @@
 @import "./govuk_style_setup";
 @import "~govuk-frontend/govuk/all";
-@import "~@ministryofjustice/frontend/moj/all";
+@import "~@ministryofjustice/frontend/moj/all";
```
Comment (darokel) [inline app/webpacker/styles/application.scss:3]: line ending

File: app/controllers/support/providers_controller.rb [hunk 6]
```diff
@@ -26,5 +41,17 @@ def filters
     def filter_params
       params.permit(:text_search, :page, :commit)
     end
+
+    def last_updated_provider
```
Comment (darokel) [inline app/controllers/support/providers_controller.rb:45]: this should go in a helper i think (and the included TimeFormat)

**Review by darokel — CHANGES_REQUESTED** [review]
Looking good. Left a couple of comments to get this over the line

File: app/helpers/support_time_helper.rb [hunk 7]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+module SupportTimeHelper
```
Comment (darokel) [inline app/helpers/support_time_helper.rb:3]: I think these helpers should be nested in a support directory to follow convention of other parts. `app/helpers/support/time_helper.rb` then you'd have:

```
module Support
  module TimeHelper
  end
end
```

**Review by darokel — APPROVED** [review]

**Pull request merged**